extensions/amp-bind/0.1/amp-state.js

```diff
@@ -55,23 +56,36 @@ export class AmpState extends AMP.BaseElement {
 
     const TAG = this.getName_();
 
-    toggle(this.element, false);
+    toggle(this.element, /* opt_display */ false);
     this.element.setAttribute('aria-hidden', 'true');
 
-    const children = this.element.children;
-    if (children.length == 1) {
-      const child = children[0];
-      if (isJsonScriptTag(child)) {
-        const json = tryParseJson(children[0].textContent, e => {
-          user().error(TAG, 'Failed to parse state. Is it valid JSON?', e);
-        });
+    // Fetch JSON from endpoint at `src` attribute if it exists,
+    // otherwise parse child script tag.
+    if (this.element.hasAttribute('src')) {
+      fetchBatchedJsonFor(this.getAmpDoc(), this.element).then(json => {
         this.updateState_(json, /* opt_isInit */ true);
-      } else {
-        user().error(TAG,
-            'State should be in a <script> tag with type="application/json"');
+      }, error => {
+        user().error(TAG, error);
+      });
+      if (this.element.children.length > 0) {
+        user().error(TAG, 'Should not have children if src attribute exists.');
+      }
+    } else {
+      const children = this.element.children;
+      if (children.length == 1) {
+        const firstChild = children[0];
+        if (isJsonScriptTag(firstChild)) {
+          const json = tryParseJson(firstChild.textContent, e => {
+            user().error(TAG, 'Failed to parse state. Is it valid JSON?', e);
+          });
+          this.updateState_(json, /* opt_isInit */ true);
+        } else {
+          user().error(TAG,
+              'State should be in a <script> tag with type="application/json"');
+        }
+      } else if (children.length > 1) {
```

Comment:
Optional: change error message to "Should contain exactly one <script> child" to include the children.length == 0 case and then flip the conditional to make the larger block (children.length == 1) the else block.